### PR TITLE
gh-126084: Fix venvwlauncher to run pythonw

### LINKED ISSUE #126084: pythonw.exe from venv in 3.13 always creates console window
# Bug report

### Bug description:

In 3.13.0a4 yet another form of launcher executable was added, apparently called "venvlauncher". This latest creation is apparently unaware that pythonw (the GUI subsystem, no-console-window form of interpreter) exists and always runs the regular python.exe (CLI subsystem). Therefore any attempt to run a script in 3.13 will either attach to the current console or create a new, blank console window.

I have not tried building venvlauncher.c with a different definition of EXENAME to see if that works, but since %PROGRAMFILES%\Python313\Lib\venv\scripts\nt\venvwlauncher.exe does not contain the string "pythonw", I'm pretty sure my guess is correct.

There is plenty of infrastructure in the Windows build to make this work, not least this block in venvlauncher.c:

```
// This should always be defined when we build for real,
// but it's handy to have a definition for quick testing
#ifndef EXENAME
#define EXENAME L"python.exe"
#endif
```

Someone probably forgot to hook up the external definition in the build.

### CPython versions tested on:

3.13

### Operating systems tested on:

Windows

### Linked PRs
* gh-126088
* gh-126142

### PR DESCRIPTION
**Pull request by chrullrich**
Fixes #126084

* Issue: gh-126084

### CODE REVIEW

**Comment by ZeroIntensity** [comment]
This is user-facing, so it needs a NEWS entry, per the bot. Probably needs a test too.

Also, for future reference, don't force push--it makes the review process more difficult. Everything is squash merged at the end anyway.

**Comment by chrullrich** [comment]
That test works fine locally. What could be different about the CI environment?

**Comment by chrullrich** [comment]
So it's not the realpath(). I don't want to waste massive amounts of time and energy by probing around the CI environment and running the whole lot each time. Any suggestions?

**Comment by chrullrich** [comment]
> Locally for me, `_winapi.GetModuleFileName(0)` always returns the path to the Python process that launched the subprocess. 

It is supposed to return the path to the executable for the _current_ process.

> Would `sys.executable` work?

No:

    C:\Daten>pyv\x64\Scripts\python.exe
    Python 3.13.0 (tags/v3.13.0:60403a5, Oct  7 2024, 09:38:07) [MSC v.1941 64 bit (AMD64)] on win32
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import sys; print(sys.executable)
    C:\Daten\pyv\x64\Scripts\python.exe
    >>> import _winapi; print(_winapi.GetModuleFileName(0))
    C:\Program Files\Python313\python.exe

When in a venv, `sys.executable` is the wrapper in the venv. I'm looking for the "real" Python in the installation that was used to create the venv.

One moment please ...

File: Lib/test/test_venv.py [hunk 1]
```diff
@@ -872,6 +872,24 @@ def test_venv_same_path(self):
...
+    # gh-126084: venvwlauncher should run pythonw, not python
+    @requireVenvCreate
+    @unittest.skipUnless(os.name == 'nt', 'only relevant on Windows')
+    def test_venvwlauncher(self):
+        """
+        Test that the GUI launcher runs the GUI python.
+        """
+        rmtree(self.env_dir)
+        venv.create(self.env_dir)
+        exename = "pythonw_d.exe"
```
Comment (devdanzin) [inline Lib/test/test_venv.py]: Will this work on a non-debug build?

Comment (chrullrich) [reply Lib/test/test_venv.py:884]: Just fixed that in the previous commit. What it won't work on is a free-threaded build, though.

Comment (chrullrich) [reply Lib/test/test_venv.py]: That was the original problem, actually; the GitHub CI environment _is_ a debug build (as should have been obvious), and that is why the first few attempts failed.

**Comment by zooba** [comment]
Thanks for finding and fixing this! What a silly mistake (probably mine...)

**Comment by zooba** [comment]
Doesn't apply to 3.12, only 3.13.

**Pull request merged**